### ISSUE TITLE
My aimbot perfectly aims at the target but misses completly

### ISSUE THREAD

**Issue by notgoodusename** [danielkrupinski/Osiris#1382]
i disabled spread and recoil.
Why does it miss? is csgo broken or what.
https://streamable.com/xqlmwz

**Comment by sammy6369** [comment]
How did you disable spread and recoil? I couldn't find that option.

**Comment by moonm00n** [comment]
i noticed the same thing

**Comment by notgoodusename** [comment]
> How did you disable spread and recoil? I couldn't find that option.

weapon_recoil_scale 0
weapon_accuracy_nospread 1

**Comment by Vadenez** [comment]
Thats in CSGO's console, That 100% doesn't affect Osiris workings. however that will change your spread and recoil in offline games, so make sure those are set back to default when using Osiris Aimbot in Offline games. Since both you @notgoodusename and @moonm00n are experencing this problem it could be another setting in Osiris. I would help more but I don't know enough about Osiris to help you fix the problem. Srry :P

**Comment by Vadenez** [comment]
I dont think I said this clear enough, but setting:
> weapon_recoil_scale 0
and
> weapon_accuracy_nospread 1
could atualy be creating the problem of "My aimbot perfectly aims at the target but misses completly". As Osiris might still be atempting to recoil control, but doesnt need to because of those two settings, therfore the recoil cortrol of Osiris would just be making you miss down and to the left and/or right (depending on the weapon and its recoil pattern).

So basically to fix this, put:
> weapon_recoil_scale 0
and
> weapon_accuracy_nospread 1
back to there original values.

**Comment by effex1337** [comment]
> I dont think I said this clear enough, but setting:
> 
> > weapon_recoil_scale 0
> 
> and
> 
> > weapon_accuracy_nospread 1
> 
> could atualy be creating the problem of "My aimbot perfectly aims at the target but misses completly". As Osiris might still be atempting to recoil control, but doesnt need to because of those two settings, therfore the recoil cortrol of Osiris would just be making you miss down and to the left and/or right (depending on the weapon and its recoil pattern).
> 
> So basically to fix this, put:
> 
> > weapon_recoil_scale 0
> 
> and
> 
> > weapon_accuracy_nospread 1
> 
> back to there original values.

Look at the rcs code it accounts for weapon_recoil_scale when pulling so that's not an issue.

**Comment by Vadenez** [comment]
welp then IDK what could cause it, also thx for the information @effex1337

**Comment by notgoodusename** [comment]
> > I dont think I said this clear enough, but setting:
> > > weapon_recoil_scale 0
> > 
> > 
> > and
> > > weapon_accuracy_nospread 1
> > 
> > 
> > could atualy be creating the problem of "My aimbot perfectly aims at the target but misses completly". As Osiris might still be atempting to recoil control, but doesnt need to because of those two settings, therfore the recoil cortrol of Osiris would just be making you miss down and to the left and/or right (depending on the weapon and its recoil pattern).
> > So basically to fix this, put:
> > > weapon_recoil_scale 0
> > 
> > 
> > and
> > > weapon_accuracy_nospread 1
> > 
> > 
> > back to there original values.
> 
> Look at the rcs code it accounts for weapon_recoil_scale when pulling so that's not an issue.

no, there is a variable that adjust to the recoil so this doesnt make sense 
        const static auto weaponRecoilScale{ interfaces.cvar->findVar("weapon_recoil_scale") };
        auto aimPunch{ localPlayer->aimPunchAngle() * weaponRecoilScale->getFloat() };
at aimbot.cpp

**Comment by xAkiraMiura** [comment]
it could be the backtrack accidently shifting to the wrong tick? if you disable backtrack does it hit perfectly or still miss.

**Comment by notgoodusename** [comment]
> it could be the backtrack accidently shifting to the wrong tick? if you disable backtrack does it hit perfectly or still miss.

still misses

**Comment by mdilai** [comment]
Lambda function you need:
```
        constexpr auto velocityExtrapolate {
            [](Entity* entity, const Vector& destination) constexpr noexcept {
                return destination + (entity->velocity() * memory->globalVars->intervalPerTick);
            }
        };
```
Example of usage:
```
                if (fov < bestFov) {
                    bestFov = fov;
                    bestTarget = config->aimbot[weaponIndex].velocityExtrapolation ? velocityExtrapolate(entity, bonePosition) : bonePosition;
                    bestAngle = angle;
                }
```

And you can look at my full re-implementation of Osiris aimbot at https://github.com/mdilai/JweegaCSGO/blob/master/Jweega/Hacks/Aimbot.cpp

**Comment by notgoodusename** [comment]
> Lambda function you need:
> 
> ```
>         constexpr auto velocityExtrapolate {
>             [](Entity* entity, const Vector& destination) constexpr noexcept {
>                 return destination + (entity->velocity() * memory->globalVars->intervalPerTick);
>             }
>         };
> ```
> 
> Example of usage:
> 
> ```
>                 if (fov < bestFov) {
>                     bestFov = fov;
>                     bestTarget = config->aimbot[weaponIndex].velocityExtrapolation ? velocityExtrapolate(entity, bonePosition) : bonePosition;
>                     bestAngle = angle;
>                 }
> ```
> 
> And you can look at my full re-implementation of Osiris aimbot at https://github.com/mdilai/JweegaCSGO/blob/master/Jweega/Hacks/Aimbot.cpp

i have done just that on my project and i am using it in the video
